streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -74,10 +74,172 @@ protected Topology(final InternalTopologyBuilder internalTopologyBuilder) {
      * {@link #addSource(AutoOffsetReset, String, String...) adding a source processor} or when creating {@link KStream}
      * or {@link KTable} via {@link StreamsBuilder}.
      */
```

Comment:
We also need to add a deprecation tag to the JavaDocs.

```
@deprecated Since 4.0. Use {@link org.apache.kafka.streams.AutoOffsetReset} instead.
```